landing-page/content/common/statistics-format-spec.md

```diff
@@ -0,0 +1,111 @@
+format designed to store information such as statistics about data managed in an
+Iceberg table that cannot be stored directly within the Iceberg manifest. A
+statistics file contains arbitrary pieces of information (here called "blobs"),
+along with metadata necessary to interpret them.
+
+## Format specification
+
+A file conforming to the format specification should have the structure as
+described below.
+
+### File structure
+
+The file has the following structure
+
+```
+Magic Blob₁ Blob₂ ... Blobₙ Footer
+```
+
+where
+
+- `Magic` is four bytes 0x80, 0x70, 0x73, 0x83 (short for: Plain Format for
+  Iceberg Statistics),
+- `Blobᵢ` is i-th blob contained in the file, to be interpreted by application
+  according to the footer,
+- `Footer` is defined below.
+
+### Footer structure
+
+Footer has the following structure
+
+```
+FooterPayload FooterPayloadSize Reserved FileFormatVersion Magic
+```
+
+where
+
+- `FooterPayload` LZ4-compressed, UTF-8 encoded JSON payload describing the
+  blobs in the file, with the structure described below,
+- `FooterPayloadSize` is a length in bytes of the `FooterPayload`, stored as 4
+  byte integer, little-endian,
+- `Reserved` is 4 bytes reserved for future use, currently should be written as
+  0x00, 0x00, 0x00, 0x00,
+- `FileFormatVersion` is a number, stored as 4 byte integer, little-endian.
+
+### Footer Payload
+
+Footer payload bytes is LZ4-compressed, UTF-8 encoded JSON payload representing
+a single `FileMetadata` object.
+
+#### FileMetadata
+
+`FileMetadata` has the following fields
+
+
+| Field Name | Field Type             | Required | Description |
+| ---------- | ---------------------- | -------- | ----------- |
+| blobs      | list of Blob objects   | yes      |
+| properties | map {string: string}   | no       | storage for arbitrary meta-information, like writer identification/version
+
+#### Blob
+
+`Blob` has the following fields
+
+| Field Name | Field Type             | Required | Description |
+| ---------- | ---------------------- | -------- | ----------- |
+| type       | string                 | yes      | See [Blob types](#blob-types)
+| columns    | list of integer numbers| yes      | list of column IDs the blob was computed for
+| offset     | integer number (long)  | yes      | The offset in the file where the blob contents start. Reader should assume the value can be more than 2^32.
+| length     | integer number         | yes      | The length of the blob stored in the file
+| compression_codec | string          | no       | See [Compression codecs](#compression-codecs). If omitted, the data is assumed to be uncompressed.
+
+### Blob types
+
+The file format makes no assumptions about the information stored in it. The
+applications using the format need to agree on type of information stored, the
+format of the information and type designators. The list below is not part of
+the file format specification itself, but is provided for readers' convenience.
```

Comment:
I would be a bit more strict here. The file format requirements are that readers ignore any blobs that they do not know how to use. But we still way to make this the canonical place to standardize the blobs.

Reply:
I agree we want the blobs format to be standardized. However, there is nothing preventing implementors from adding their own blob formats. For example, a query engine's optimizer can have a unique optimization feature that works but only if you provide the HLL  sketch with specific hash functions. This may be odd for general Iceberg community, but be right for this particular engine. Of course, later when the usefulness is validated, the blob type may become standardized.

Thus, to me, the blob types is more of a supplemental information. The format itself doesn't define them, is open. The project and the processes around it enforce that we know what blob types we uses.

Reply:
see also @jh3507's comment in https://docs.google.com/document/d/1we0BuQbbdqiJS2eUFC_-6TPSuO57GXivzKmcTzApivY/edit?disco=AAAAWq4ma7Q

Reply:
There isn't anything preventing it, but I think we should still state in the spec that they _should_ be documented here. Stating that you can define your own invites fragmentation, so I'd try to require that the formats are registered. People could ignore it, but at least we will have tried to maintain a standard set.

Reply:
Sure, i reworded this. And this became very short thanks to this.